### PR TITLE
 PLA-196 Unify integration protos

### PR DESCRIPTION
**Pull request by rhvivancoeffio**
# Proposed Changes
Unify protos for integrations

### Impacted Routes

GetDeviceDefinitionByID
GetDeviceDefinitionByMMY
GetIntegrations
GetDeviceDefinitionIntegration

### CODE REVIEW

**Review by JamesReate — COMMENTED** [review]
discussed some ideas via DM to unify things further and reduce duplication

File: pkg/grpc/device_definition.proto [hunk 1]
```diff
@@ -17,62 +17,40 @@ message GetDeviceDefinitionItemResponse {
...
-
-	message DeviceStyles {
-		string id = 1;
-		string source = 2;
-		string name = 3;
-		string subModel = 4;
-		string deviceDefinitionId = 5;
-		string external_style_id = 6;
-	}
+message DeviceType {
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:30]: Trying to catch up on this—a "device type" is more or less the mmy? What's the `type` field?

File: pkg/grpc/device_definition.proto [hunk 2]
```diff
@@ -17,62 +17,40 @@ message GetDeviceDefinitionItemResponse {
...
+	string subModel = 4;
+	string deviceDefinitionId = 5;
+	string external_style_id = 6;
 }
 
+message DeviceMake {
+	string id = 1;
+	string name = 2;
+	string logo_url = 3;
+	string oem_platform_name = 4;
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:51]: What is this?

File: pkg/grpc/device_definition.proto [hunk 3]
```diff
@@ -17,62 +17,40 @@ message GetDeviceDefinitionItemResponse {
...
+}
 
-	message Make {
-		string id = 1;
-		string name = 2;
-		string logo_url = 3;
-		string oem_platform_name = 4;
-		uint64 token_id = 5;
-	}
+message DeviceStyles {
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:38]: Why is this plural?

File: pkg/grpc/device_definition.proto (hunk 1 above)
Comment (JamesReate) [reply pkg/grpc/device_definition.proto:30]: it's a carryover from the rest endpoint where we'd return the MMY stuff in a separate DeviceType object to make it more generic although everything is basically vehicles. I'm fine getting rid of it if it doesn't make sense anymore.

File: pkg/grpc/device_definition.proto [hunk 4]
```diff
@@ -17,62 +17,40 @@ message GetDeviceDefinitionItemResponse {
...
+	string id = 1;
+	string source = 2;
+	string name = 3;
+	string subModel = 4;
+	string deviceDefinitionId = 5;
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:43]: We mix casing styles here a bit—is this intentional?

File: pkg/grpc/device_definition.proto (hunk 2 above)
Comment (JamesReate) [reply pkg/grpc/device_definition.proto:51]: `oem_platform_name` is like FordPass, OnStart, etc frontends wanted it at some point.

File: pkg/grpc/device_definition.proto (hunk 1 above)
Comment (elffjs) [reply pkg/grpc/device_definition.proto:30]: Oh so `type = "vehicle"` or something?

File: pkg/grpc/device_definition.proto [hunk 5]
```diff
@@ -95,42 +73,37 @@ message GetDeviceDefinitionByMMYRequest {
 }
 
 message GetIntegrationResponse {
-	repeated GetIntegrationItemResponse integrations = 1;
+	repeated DeviceIntegration integrations = 1;
 }
 
 message EmptyRequest
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:79]: [`google.protobuf.Empty`](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Empty) might interest you.

File: pkg/grpc/device_definition.proto [hunk 6]
```diff
@@ -95,42 +73,37 @@ message GetDeviceDefinitionByMMYRequest {
...
-	GetAutoPiPowertrainTemplate auto_pi_powertrain_template = 6;
-
-	message  GetAutoPiPowertrainTemplate {
-		int32 BEV = 1;
-		int32 HEV = 2;
-		int32 ICE = 3;
-		int32 PHEV = 4;
-	}
+	string region = 5;
+	string country = 6;
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:90]: Do we do this by country anymore? I thought this stopped.

File: pkg/grpc/device_definition.proto [hunk 7]
```diff
@@ -95,42 +73,37 @@ message GetDeviceDefinitionByMMYRequest {
...
+	string capabilities = 7;
+	string deviceDefinitionId = 8;
+	int32 auto_pi_default_template_id = 9;
+	GetAutoPiPowertrainTemplate auto_pi_powertrain_template = 10;
```
Comment (elffjs) [inline pkg/grpc/device_definition.proto:94]: Does the `Get` on the front make sense?

**Pull request merged**